lib/grid.rb

```diff
@@ -1,25 +1,50 @@
+require_relative 'cell.rb'
+
 class Grid
-  attr_accessor :structure
+  attr_accessor :cells, :computer_options
 
   def initialize
-    @structure = grid_structure
+    @cells = build_grid
+    @computer_options = []
+  end
+
+  def mark_with_player_choice(valid_choice, marker)
+    mark(valid_choice, marker)
   end
 
-  def mark(coordinate)
-    structure[coordinate] = 'X'
+  def mark(coordinate, value)
```

Comment:
You can see here that the cell are just literal data structures and this class has to know way to much about the way that the data exists and is stored. That isn't good encapsulation.